### PR TITLE
Multisig

### PR DESCRIPTION
**Pull request by sullof**
This implements MultiSignature.
To stay compatible with the current requirements, `trx#sign` expects the new boolean `multisig` if a signature is required for a transaction already signed or that we want to sign with a private key different from the owner_address one.

### CODE REVIEW

File: src/lib/trx.js [hunk 1]
```diff
@@ -570,20 +570,29 @@ export default class Trx {
...
         if(utils.isFunction(useTronHeader)) {
             callback = useTronHeader;
             useTronHeader = true;
+            multisig = false;
+        }
+
+        if(utils.isFunction(privateKey)) {
+            callback = privateKey;
+            privateKey = this.tronWeb.defaultPrivateKey;
+            useTronHeader = true;
```
Comment (Kondax) [inline src/lib/trx.js]: No need to overwrite `useTronHeader` or `multisig`, as they'll already be defaulted to `true` and `false` respectively from the above function header.

File: src/utils/crypto.js [hunk 2]
```diff
@@ -56,7 +56,12 @@ export function signTransaction(priKeyBytes, transaction) {
     const txID = transaction.txID;
     const signature = ECKeySign(hexStr2byteArray(txID), priKeyBytes);
 
-    transaction.signature = [signature];
+    if (Array.isArray(transaction.signature)) {
+        if (!~transaction.signature.indexOf(signature))
```
Comment (Kondax) [inline src/utils/crypto.js]: It's better to use `transaction.signature.includes(signature)`, which is more readable

**Comment by Kondax** [comment]
I've written some general comments, however the overall implementation looks fine.

A suggestion for you @sullof, maybe it would be a good idea to create a `develop` branch and merge PRs into that branch? You can then bump the version and distribute to npm, which would let you tag a release and then merge into master. This might be cleaner than having code changes to master that do not bump the version / aren't released to npm.

_I wrote this just before you merged the PR by the way, so apologies for not commenting in time_

**Comment by sullof** [comment]
@Kondax I merged it before seeing your comment. I am doing a new PR with those changes (I agree on them).

**Comment by sullof** [comment]
I also agree about the develop branch, by the way.

**Pull request merged**